WalletWasabi/Extensions/NBitcoinExtensions.cs

```diff
@@ -456,29 +456,19 @@ public static FeeRate GetSanityFeeRate(this MemPoolInfo me)
 			return sanityFee;
 		}
 
-		public static int EstimateOutputVsize(this Script script)
-		{
-			if (script.IsScriptType(ScriptType.P2WPKH))
-			{
-				return Constants.OutputSizeInBytes;
-			}
-			else
+		public static int EstimateOutputVsize(this Script scriptPubKey) =>
+			StandardScripts.IsStandardScriptPubKey(scriptPubKey) switch
 			{
-				throw new NotImplementedException($"Weight estimation isn't implemented for provided script type.");
-			}
-		}
+				true => new TxOut(Money.Zero, scriptPubKey).GetSerializedSize(),
+				false => throw new NotImplementedException($"Size estimation isn't implemented for provided script type.")
+			};
```

Comment:
FWIW in #5531 i made this just return the output size just like you did here, but without the standardness check, why is that appropriate?

Reply:
It shouldn't be here.